tree/20_cfe_basics/log_rudder.cf

```diff
@@ -516,11 +516,13 @@ bundle agent _method_reporting_context_v4(c_name, c_key, report_id)
     "report_data.report_id_r"    string => "${report_id}";
     # Canonified version, to be used in policies (as class prefix)
     "report_data.report_id"      string => canonify("${report_id}");
-    "report_data.method_id"      string => canonify("${report_id}");
+    "report_data.method_id"      string => canonify("${report_id}_${report_data.directive_id}");
     "report_data.identifier"     string => "${report_data.rule_id}@@${report_data.directive_id}@@${report_data.report_id_r}";
 
   methods:
-    "${report_id}" usebundle => rudder_method_id_reset;
```

Comment:
I think the promiser should be report_data.method_id or else it won't be rexecuted if it has already been done once for this report_id